crates/router/src/connector/utils.rs

```diff
@@ -414,6 +422,27 @@ impl AddressDetailsData for api::AddressDetails {
     }
 }
 
+pub trait MandateData {
+    fn get_end_date(&self, format: date_time::DateFormat) -> Result<String, Error>;
+    fn get_meta_data(&self) -> Result<pii::SecretSerdeValue, Error>;
+}
+
+impl MandateData for payments::MandateAmountData {
+    fn get_end_date(&self, format: date_time::DateFormat) -> Result<String, Error> {
+        let date = self.end_date.ok_or_else(missing_field_err(
+            "mandate_data.mandate_type.{multi_use|single_use}.end_date",
+        ))?;
+        date_time::format_date(date, format)
+            .into_report()
+            .change_context(errors::ConnectorError::DateFormattingFailed)
+    }
+    fn get_meta_data(&self) -> Result<pii::SecretSerdeValue, Error> {
+        self.meta_data.clone().ok_or_else(missing_field_err(
```

Comment:
Hey, as @jarnura mentioned if you don't have it as a trait, can use also make the return value as reference, let's try to avoid clone, if possible